File: test_exa_api.py
import requests
import json

# Test the Exa.ai search API
def test_exa_api():
    print("🔍 Testing Exa.ai Search API\n")
    
    url = "https://api.exa.ai/search"
    api_key = "96e0fcca-9781-4785-b71e-c77ed653f168"
    
    headers = {
        "content-type": "application/json",
        "x-api-key": api_key
    }
    
    payload = {
        "query": "Latest news on Nvidia",
        "numResults": 10,
        "type": "auto",
        "contents": {
            "highlights": {
                "maxCharacters": 4000
            }
        }
    }
    
    print("Request Details:")
    print(f"URL: {url}")
    print(f"API Key: {api_key[:8]}...{api_key[-4:]}")
    print(f"Query: {payload['query']}")
    print(f"Number of results: {payload['numResults']}")
    print()
    
    try:
        response = requests.post(url, headers=headers, json=payload, timeout=30)
        
        print(f"Status Code: {response.status_code}")
        print(f"Response Headers: {dict(response.headers)}")
        
        if response.status_code == 200:
            print("✅ Success! API is working")
            data = response.json()
            
            # Analyze the response
            if 'results' in data:
                results = data['results']
                print(f"Number of results returned: {len(results)}")
                
                if results:
                    print("\nFirst result preview:")
                    first_result = results[0]
                    print(f"Title: {first_result.get('title', 'N/A')}")
                    print(f"URL: {first_result.get('url', 'N/A')}")
                    print(f"Score: {first_result.get('score', 'N/A')}")
                    
                    if 'highlights' in first_result:
                        highlights = first_result['highlights']
                        print(f"Highlights: {highlights[:200]}..." if len(highlights) > 200 else f"Highlights: {highlights}")
                else:
                    print("No results found")
            else:
                print("Unexpected response format:")
                print(json.dumps(data, indent=2)[:500])
                
        else:
            print(f"❌ API Error: {response.status_code}")
            print(f"Error message: {response.text}")
            
    except requests.exceptions.RequestException as e:
        print(f"❌ Request failed: {str(e)}")
    except Exception as e:
        print(f"❌ Unexpected error: {str(e)}")

# Run the test
test_exa_api()

File: test_exa_api.js
import fetch from 'node-fetch';

// Test the Exa.ai search API with JavaScript
async function testExaAPI() {
    console.log('🔍 Testing Exa.ai Search API (JavaScript)\n');
    
    const url = 'https://api.exa.ai/search';
    const apiKey = '96e0fcca-9781-4785-b71e-c77ed653f168';
    
    const headers = {
        'content-type': 'application/json',
        'x-api-key': apiKey
    };
    
    const payload = {
        "query": "Latest news on Nvidia",
        "numResults": 10,
        "type": "auto",
        "contents": {
            "highlights": {
                "maxCharacters": 4000
            }
        }
    };
    
    console.log('Request Details:');
    console.log(`URL: ${url}`);
    console.log(`API Key: ${apiKey.substring(0, 8)}...${apiKey.substring(apiKey.length - 4)}`);
    console.log(`Query: ${payload.query}`);
    console.log(`Number of results: ${payload.numResults}`);
    console.log();
    
    try {
        const response = await fetch(url, {
            method: 'POST',
            headers: headers,
            body: JSON.stringify(payload)
        });
        
        console.log(`Status Code: ${response.status}`);
        console.log(`Status Text: ${response.statusText}`);
        
        if (response.ok) {
            console.log('✅ Success! API is working');
            const data = await response.json();
            
            // Analyze the response
            if (data.results && Array.isArray(data.results)) {
                console.log(`Number of results returned: ${data.results.length}`);
                
                if (data.results.length > 0) {
                    console.log('\nFirst result preview:');
                    const firstResult = data.results[0];
                    console.log(`Title: ${firstResult.title || 'N/A'}`);
                    console.log(`URL: ${firstResult.url || 'N/A'}`);
                    console.log(`Score: ${firstResult.score || 'N/A'}`);
                    
                    if (firstResult.highlights) {
                        const highlights = firstResult.highlights;
                        console.log(`Highlights: ${highlights.length > 200 ? highlights.substring(0, 200) + '...' : highlights}`);
                    }
                } else {
                    console.log('No results found');
                }
            } else {
                console.log('Unexpected response format:');
                console.log(JSON.stringify(data, null, 2).substring(0, 500));
            }
            
        } else {
            console.log(`❌ API Error: ${response.status}`);
            const errorText = await response.text();
            console.log(`Error message: ${errorText.substring(0, 200)}`);
        }
        
    } catch (error) {
        console.error(`❌ Request failed: ${error.message}`);
    }
}

// Run the test
await testExaAPI();